### ISSUE TITLE
UPUnsupportedProblemTypeError: implies keyword

### ISSUE THREAD

**Issue by MFaisalZaki** [aiplan4eu/unified-planning#610]
The parser throws due to an `implies` keyword:

![Screenshot from 2024-06-11 16-35-07](https://github.com/aiplan4eu/unified-planning/assets/9129265/644f7371-0a35-4802-a305-d6286e5d321c)

We can use Compliers to simplify such expressions. However, I can't figure out which compiler I should use.

Here is the [planning task](https://github.com/user-attachments/files/15791542/problem.zip) that throws this exception. 

Thanks

**Comment by alvalentini** [comment]
Hi @MFaisalZaki, can you also provide the python code to reproduce the error?

**Comment by MFaisalZaki** [comment]
Hi @alvalentini, I tried to provide a code to reproduce the bug since the original code that generated the bug is part of my research, and I cannot share it for now. 

My code follows the following pattern:
```
from unified_planning.io import PDDLReader
from unified_planning.engines import PlanGenerationResultStatus as ResultsStatus
from unified_planning.shortcuts import OneshotPlanner, AnytimePlanner

from unified_planning.shortcuts import *
import up_symk

domainfile = "domain.pddl"
problemfile = "p01.pddl"

task = PDDLReader().parse_problem(domainfile, problemfile)
planlist = []
k = 5
search_config = f"symq-bd(plan_selection=top_k(num_plans={k},dump_plans=true),quality=1.0)"
with AnytimePlanner(name='symk-opt', params={"symk_anytime_search_config": search_config}) as planner:
    for i, result in enumerate(planner.get_solutions(task)):
        if result.status == ResultsStatus.INTERMEDIATE:
            planlist.append(result.plan) if i < k else None

task = PDDLReader().parse_problem(domainfile, problemfile)

with Compiler(problem_kind = task.kind, 
                      compilation_kind = CompilationKind.QUANTIFIERS_REMOVING) as quantifiers_remover:
            qr_result  = quantifiers_remover.compile(task, CompilationKind.QUANTIFIERS_REMOVING)

with Compiler(problem_kind = qr_result.problem.kind, 
                compilation_kind = CompilationKind.GROUNDING) as grounder:
    gr_result = grounder.compile(qr_result.problem, CompilationKind.GROUNDING)

# perform other stuff
```
The pattern is that I solve the planning task using `symk` and then generate plans and perform other operations on it. The bug appears in the grounding phase. 

I hope this helps. If not, please feel free to close the issue until I release my original code, and then we can reopen it again.

Thanks.

**Comment by MFaisalZaki** [comment]
Hi again @alvalentini,

I could reproduce the bug using one of the publicly available packages.
![Screenshot from 2024-06-13 10-19-19](https://github.com/aiplan4eu/unified-planning/assets/9129265/0a47c4d5-3942-4b96-acb6-aa0e4dbaa96d)

Please install this Python package to reproduce this:
```
pip install git+https://github.com/pyPMT/pyPMT.git
``` 

Using the same domain and problem files provided earlier with this code snippet, you can reproduce the bug.
```
from unified_planning.io import PDDLReader
from unified_planning.engines import PlanGenerationResultStatus as ResultsStatus
from unified_planning.shortcuts import OneshotPlanner, AnytimePlanner

from unified_planning.shortcuts import *
from pypmt.shortcuts import EncoderSequential
import up_symk

domainfile = "domain.pddl"
problemfile = "p01.pddl"

task = PDDLReader().parse_problem(domainfile, problemfile)
planlist = []
k = 5
search_config = f"symq-bd(plan_selection=top_k(num_plans={k},dump_plans=true),quality=1.0)"
with AnytimePlanner(name='symk-opt', params={"symk_anytime_search_config": search_config}) as planner:
    for i, result in enumerate(planner.get_solutions(task)):
        if result.status == ResultsStatus.INTERMEDIATE:
            planlist.append(result.plan) if i < k else None

encoded_task = EncoderSequential(task)
```

**Comment by MFaisalZaki** [comment]
Also, you can try it without the need for an external package.

```
from unified_planning.io import PDDLReader

domainfile = "domain.pddl"
problemfile = "p01.pddl"

task = PDDLReader().parse_problem(domainfile, problemfile)

with Compiler(problem_kind = task.kind, 
    compilation_kind = CompilationKind.QUANTIFIERS_REMOVING) as quantifiers_remover:
    qr_result  = quantifiers_remover.compile(task, CompilationKind.QUANTIFIERS_REMOVING)

with Compiler(problem_kind = qr_result.problem.kind, 
    compilation_kind = CompilationKind.GROUNDING) as grounder:
    gr_result = grounder.compile(qr_result.problem, CompilationKind.GROUNDING)

```
[problemfiles.zip](https://github.com/user-attachments/files/15818173/problemfiles.zip)

**Comment by alvalentini** [comment]
@MFaisalZaki I tried to reproduce the bug but I was not able.

Looking at the error message, the `UPUnsupportedProblemTypeError` exception is probably raised by pyperplan, that is not able to handle implies. So it is an expected exception if you are using pyperplan with this kind of problem.

However the code that you provided should not select pyperplan to ground the problem. And if I try to run your code, pyperplan is not selected and I get no errors.

Which version of the library are you using? And which versions of the engines?

**Comment by alvalentini** [comment]
Ok, the problem is the one you reported in #593 and that is solved in #594.

The fix it is still not in a stable release but if you install the latest version of the library it should be fixed.

You can install the `master` of this repo or the latest pre release version from pypi (`pip3 install --pre unified-planning`)

**Comment by MFaisalZaki** [comment]
great, thanks @alvalentini.
